python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
I think that it can be simpler with waht we discussed in the PR

If the value is None, only return true it if the parameter is optional. That is, the "optional" means here "acceptsNone". And  we do not handle the default at all, it should be there to be used by the setDefault method, which should be called by the code that is preparing the algorithm and setting the values of the parameters.

That will be clear and fine for all uses, i think